src/main/AuthenticationRequest.ts

```diff
@@ -147,14 +147,24 @@ export default class AuthenticationRequest {
       }
     }
 
-    const regObj = verPayload.registration_uri ? (await getWithUrl(verPayload.registration_uri)) as unknown as RPRegistrationMetadataPayload : verPayload.registration;
+    const regObj: RPRegistrationMetadataPayload = verPayload.registration_uri ? remoteRegObj : verPayload.registration;
 
     if (regObj && !validateRPRegistrationMetadata(regObj)) {
       throw new Error('Registration data validation error: ' + JSON.stringify(validateRPRegistrationMetadata.errors));
     } else if (regObj?.subject_syntax_types_supported && regObj.subject_syntax_types_supported.length == 0) {
       throw new Error(`${SIOPErrors.VERIFY_BAD_PARAMS}`);
     }
   }
+
+  static async getRemoteRegistrationObj (registrationUri: string): Promise<RPRegistrationMetadataPayload> {
+    let response: RPRegistrationMetadataPayload;
+    if (registrationUri) {
```

Comment:
Why is registrationUri not marked optional? BTW, I rather would see a method that either accepts the object and the url as 2 params and always returns the object. So that you can supply the byValue and byRef params to it and it gives you back the object. Now it is calling this method with a possible null arg and then later on you inspect whether it was by ref or by value and then construct the eventual object.

Reply:
fixed